csfieldguide/static/interactives/dictionary-compression/js/dictionary-compression.js

```diff
@@ -0,0 +1,342 @@
+var codes = [];
```

Comment:
If we choose to compress an empty string we get shown 'NaN' for the percentage of text compressed

![image](https://user-images.githubusercontent.com/16066822/56999627-15a03280-6c04-11e9-93dd-bf066c5ca434.png)

Reply:
If I repeat these Chinese characters it no longer highlights any of the coded words in the dicitonary

Before adding repeated Chinese characters:
![image](https://user-images.githubusercontent.com/16066822/56999915-65cbc480-6c05-11e9-8c37-46f8c724a4e4.png)

After adding repeated Chinese characters:
![image](https://user-images.githubusercontent.com/16066822/56999892-387f1680-6c05-11e9-919e-0fa39572f859.png)

Reply:
so this problem was actually worse than this! It broke if any code was repeated in the dictionary, e.g. if the dictionary already contained `test` and you added `testtest` it would break, ahh! Fixed now.